### PR TITLE
Change where we store application data

### LINKED ISSUE #298: Change data directories to make more sense
We're currently (transitively) using [xdg-portable](https://www.npmjs.com/package/xdg-portable) for managing our paths; this is leading to unusual things like `%LOCALAPPDATA%\xdg.state\rancher-desktop\logs` or `~/Library/State/rancher-desktop`, which looks foreign on both systems.  We may want to consider fixing things to use more native-looking paths. instead.

### PR DESCRIPTION
**Pull request by mook-as**
This changes where we store our application data; fixes #298 

Directories (unchanged directories marked by†):

Type | Current Location | New (Windows) | New (Mac)
--- | --- | --- | ---
Logs | `${XDG.state}/logs` | `LocalAppData\rancher-desktop\logs` | `~/Library/Logs/rancher-desktop`
Lima VM | `${XDG.state}/lima` | N/A | `~/Library/Application Support/rancher-desktop/lima`
WSL distro | `${XDG.state}/distro` | `LocalAppData\rancher-desktop\distro` | N/A
Caches | `${XDG.cache}` | `LocalAppData\rancher-desktop\cache` | `~/Library/Caches/rancher-desktop`
Settings | `${XDG.config}` | `RoamingAppData\rancher-desktop` | `~/Library/Preferences/rancher-desktop`†
Electron / Chromium Data | `${XDG.data}` | `RoamingAppData\rancher-desktop`† | `~/Library/Application Support/rancher-desktop`†

### CODE REVIEW

File: src/main/paths.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    console.log(`Migrated ${ info } data from ${ oldPath } to ${ newPath }`);
+
+    return true;
+  } catch (ex) {
+    if (!['ENOENT', 'EEXIST'].includes(ex.code)) {
+      console.error(`Error moving ${ info }:`, ex);
+    } else {
+      console.log(`Could not move ${ oldPath } to ${ newPath }:`, ex);
+    }
+    recursiveRemoveSync(oldPath);
```
Comment (ericpromislow) [inline src/main/paths.ts]: Why aren't you also copying the oldPath to the new before deleting it?

**Review by ericpromislow — APPROVED** [review]
went through everything, looks good now

File: src/main/paths.ts (hunk 1 above)
Comment (mook-as) [reply src/main/paths.ts]: We're not doing any fancy things here (we're managing both the old and new paths), so there isn't going to be any cross-device stuff.  We really want this to be fast (it's blocking the startup path).

Comment (ericpromislow) [reply src/main/paths.ts]: Thought so, and I approved the PR. No need to change this.

**Pull request merged**